### PR TITLE
cmd: add decoder feature

### PR DESCRIPTION
**Pull request by imtbkcat**
I implement a simple tool to deocde mvcc key information. It currently support four format.

- table_row key format, like this: 
  `t\x80\x00\x00\x00\x00\x00\x07\x8f_r\x80\x00\x00\x00\x00\x08\x3b\xba`
 ```
 ./tidb-ctl decoder -f=table_row -k "t\x80\x00\x00\x00\x00\x00\x07\x8f_r\x80\x00\x00\x00\x00\x08\x3b\xba"
table_id: 1935
row_id: 539578
```
- table_index format, like this:
 `t\x80\x00\x00\x00\x00\x00\x00\x5f_i\x80\x00\x00\x00\x00\x00\x00\x01\x03\x80\x00\x00\x00\x00\x00\x00\x02\x03\x80\x00\x00\x00\x00\x00\x00\x02`
```
./tidb-ctl decoder -f=table_index -k "t\x80\x00\x00\x00\x00\x00\x00\x5f_i\x80\x00\x00\x00\x00\x00\x00\x01\x03\x80\x00\x00\x00\x00\x00\x00\x02\x03\x80\x00\x00\x00\x00\x00\x00\x02"
table_id: 95
index_id: 1
type: Int64, value: 2
type: Int64, value: 2
```
- base64 value format, like this:
`CAQCBmFiYw==`
```
./tidb-ctl decoder -f=value -k "CAQCBmFiYw=="
type: Int64, value: 2
type: Bytes, value: abc
```

document will be finished soon.

### CODE REVIEW

**Comment by imtbkcat** [comment]
PTAL @jackysp

**Comment by shenli** [comment]
Good job!

**Comment by shenli** [comment]
Please update the document.

File: cmd/base64decode.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
+// Copyright 2017 PingCAP, Inc.
```
Comment (jackysp) [inline cmd/base64decode.go]: 2017 -> 2018

File: cmd/decoder.go [hunk 2]
```diff
@@ -0,0 +1,176 @@
+// Copyright 2017 PingCAP, Inc.
```
Comment (jackysp) [inline cmd/decoder.go]: 2017 -> 2018

File: doc/tidb-ctl_decoder.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
+## tidb-ctl deocder
```
Comment (jackysp) [inline doc/tidb-ctl_decoder.md]: deocder -> decoder

File: cmd/decoder.go [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+// decoderCmd represents the key-decoder command
+var decoderCmd = &cobra.Command{
+	Use:   "decoder -f <key-format> -k <key>",
+	Short: "decode tabel_row/table_index/value format key to readable format",
+	Long:  "decode tabel_row/table_index/value format key to readable format",
+	RunE:  decodeKeyFunc,
+}
+
+var indexTypeToString = map[byte]string{
```
Comment (jackysp) [inline cmd/decoder.go]: If the type changed, we need to update it. Is it possible to avoid it?

File: doc/tidb-ctl_base64decode.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
+## tidb-ctl base64decode
+
+deocde base64 value to uint64
```
Comment (jackysp) [inline doc/tidb-ctl_base64decode.md]: The documents could be generated by `make doc`.

File: cmd/decoder.go (hunk 4 above)
Comment (imtbkcat) [reply cmd/decoder.go]: How about putting this map into TiDB? @jackysp

**Comment by imtbkcat** [comment]
addressed @jackysp

**Review by jackysp — COMMENTED** [review]
Fix deocde -> decode, there is not only one typo in the pr. The docs could be generated by `make doc`.

**Comment by imtbkcat** [comment]
addressed @jackysp

**Review by jackysp — APPROVED** [review]
LGTM

**Pull request merged**